src/hdr_histogram.c

```diff
@@ -386,6 +388,58 @@ int hdr_calculate_bucket_config(
     return 0;
 }
 
+int yb_hdr_calculate_bucket_config(
+        int64_t lowest_discernible_value,
+        int64_t highest_trackable_value,
+        int yb_bucket_factor,
+        struct hdr_histogram_bucket_config* cfg)
+{
+    int32_t sub_bucket_count_magnitude;
+
+    // YB change: hardcoded sig figs to 1 for memory performance, controlling bucket sizes with yb_bucket_factor
+    cfg->lowest_discernible_value = lowest_discernible_value;
+    cfg->significant_figures = YB_SIG_FIGS;
+    cfg->highest_trackable_value = highest_trackable_value;
+    typedef enum {EIGHT = 8, SIXTEEN = 16, THIRTY_TWO = 32} valid_bucket_factor;
+    valid_bucket_factor bf = (valid_bucket_factor) yb_bucket_factor;
+    // YB change: override subbucket count magnitude to set subbuckets per bucket equal to yb_bucket_factor variable
+    switch(bf)
+    {
+        case EIGHT:
+            __attribute__ ((fallthrough));
+        case SIXTEEN:
+            __attribute__ ((fallthrough));
+        case THIRTY_TWO:
+            sub_bucket_count_magnitude = log(yb_bucket_factor)/log(2);
+            break;
+        default:
+            return EINVAL;
+            break;
+    }
+
+    cfg->sub_bucket_half_count_magnitude = ((sub_bucket_count_magnitude > 1) ? sub_bucket_count_magnitude : 1) - 1;
+    double unit_magnitude = log((double)lowest_discernible_value) / log(2);
+    if (INT32_MAX < unit_magnitude)
+    {
+        return EINVAL;
+    }
+
+    cfg->unit_magnitude = (int32_t) unit_magnitude;
+    cfg->sub_bucket_count      = (int32_t) pow(2, (cfg->sub_bucket_half_count_magnitude + 1));
+    cfg->sub_bucket_half_count = cfg->sub_bucket_count / 2;
+    cfg->sub_bucket_mask       = ((int64_t) cfg->sub_bucket_count - 1) << cfg->unit_magnitude;
+
+    if (cfg->unit_magnitude + cfg->sub_bucket_half_count_magnitude > 61)
+    {
+        return EINVAL;
+    }
+
```

Comment:
Most of this logic is duplicated from the original function. Can we pull out the duplicated logic, and call it from both bucket config commands?